### PR TITLE
Refactor unused var in presto-pinot

### PR DESCRIPTION
**Pull request by brandboat**
#4188
This PR is a preparation before enable UnusedVariable error prone check.

### CODE REVIEW

**Comment by brandboat** [comment]
CI failed -- #2994

File: presto-pinot/src/main/java/io/prestosql/pinot/client/PinotClient.java [hunk 1]
```diff
@@ -94,10 +92,8 @@
     private static final String QUERY_URL_TEMPLATE = "http://%s/query/sql";
 
     private final List<String> controllerUrls;
-    private final PinotMetrics metrics;
```
Comment (findepi) [inline presto-pinot/src/main/java/io/prestosql/pinot/client/PinotClient.java:97]: It looks like `PinotMetrics` class is mostly unused except for two utility methods (`doWithRetries`, `isValidPinotHttpResponseCode`).

**Review by findepi — COMMENTED** [review]
Looks good to me.

@elonazoulay do you want to take a look?

File: presto-pinot/src/main/java/io/prestosql/pinot/client/PinotClient.java (hunk 1 above)
Comment (brandboat) [reply presto-pinot/src/main/java/io/prestosql/pinot/client/PinotClient.java:97]: Should we keep other unused methods ?
If we need to remove those unused methods , `PinotMetrics` and `PinotMetricsStats` could be removed (and move `doWithRetries` to `PrestoClient`).

Comment (findepi) [reply presto-pinot/src/main/java/io/prestosql/pinot/client/PinotClient.java:97]: Correct, no need to keep the unused methods.

**Comment by findepi** [comment]
Merged (25e925095cda8a2a792421c1724eb2603e764d3b), thanks!

**Pull request closed without merging**